### PR TITLE
pyre 0.9.22 (new formula)

### PR DESCRIPTION
**Pull request by Anselmoo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by Anselmoo** [comment]
@carlocab Sorry, can I ask you what is the current best approach to deal with the CPU issue

```shell
ƛ [Errno 86] Bad CPU type in executable: '/opt/homebrew/bin/pyre.bin' 
```


Locally it is working fine:

image

File: Formula/p/pyre.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  desc "Performant type-checking for python"
+  homepage "https://pyre-check.org"
+  url "https://files.pythonhosted.org/packages/fd/94/bbbb7b51310bc17fd80453f17c72fdf5079dbfe97558c511295be50a4880/pyre-check-0.9.22.tar.gz"
+  sha256 "e082f926dff71661959535c3936fca5ad40a44858b5fd3e99009a616a1b57083"
+  license "MIT"
+  head "https://github.com/facebook/pyre-check.git", branch: "main"
+
+  depends_on "rust" => :build
+  depends_on "libyaml"
+  depends_on "python@3.12"
```
Comment (chenrui333) [inline Formula/p/pyre.rb:13]: can we also do py3.13?
```suggestion
  depends_on "python@3.13"
```

Comment (Anselmoo) [reply Formula/p/pyre.rb:13]: We should test it, was a little irritated about the [tox](https://github.com/facebook/pyre-check/blob/fa94e180b696f8da3fe564fc67addf5b3e649f0c/pyproject.toml#L6-L8) classifier


```config
[tox]
envlist = py38, py39, py310, py311, py312, lint
```

but [setup.py](https://github.com/facebook/pyre-check/blob/fa94e180b696f8da3fe564fc67addf5b3e649f0c/scripts/pypi/setup.py#L99-L112) seems to be a little more off ...

```py
       classifiers=[
            "Development Status :: 5 - Production/Stable",
            "Environment :: Console",
            "Intended Audience :: Developers",
            "License :: OSI Approved :: MIT License",
            "Operating System :: MacOS",
            "Operating System :: POSIX :: Linux",
            "Programming Language :: Python",
            "Programming Language :: Python :: 3",
            "Programming Language :: Python :: 3.8",
            "Programming Language :: Python :: 3.9",
            "Topic :: Software Development",
            "Typing :: Typed",
        ],
```

File: Formula/p/pyre.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      expect "ƛ Also initialize watchman in the current directory? \\[Y/n\\]"
+      send "Y\r"
+      expect "ƛ Which directory(ies) should pyre analyze? (Default: `.`):"
+      send ".\r"
+      expect eof
+    EOS
+
+    output = shell_output("#{bin}/pyre check 2>&1", 1)
+    assert_match "ƛ Found 1 type error", output
+    assert_match "test.py:1:0 Incompatible variable type [9]: i is declared to have type `int` " \
```
Comment (Anselmoo) [inline Formula/p/pyre.rb:126]: will change to `assert_includes output` help for the Linux testing?
```suggestion
    assert_includes "Found 1 type error", output
```

File: Formula/p/pyre.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                 "but is used as type `str`.", output
```
Comment (Anselmoo) [inline Formula/p/pyre.rb:127]:
```suggestion
    assert_includes "Found 1 type error", output
    assert_match "test.py:1:0 Incompatible variable type [9]: i is declared to have type `int` " \
                 "but is used as type `str`.", output
```

**Comment by chenrui333** [comment]
> @carlocab Sorry, can I ask you what is the current best approach to deal with the CPU issue
> 
> ```shell
> ƛ [Errno 86] Bad CPU type in executable: '/opt/homebrew/bin/pyre.bin' 
> ```
> 
> Locally it is working fine:
> 
> image

this is the upstream issue, https://github.com/facebook/pyre-check/issues/614

**Comment by chenrui333** [comment]
after another look, you might need to build pyre.bin with ocaml toolchain, you can check https://github.com/nix-windows/nixpkgs/blob/dc7ea0147fc07a391d15564e5f133e81095d57f4/pkgs/development/tools/pyre/default.nix setup for ref.

**Comment by Anselmoo** [comment]
> after another look, you might need to build pyre.bin with ocaml toolchain, you can check https://github.com/nix-windows/nixpkgs/blob/dc7ea0147fc07a391d15564e5f133e81095d57f4/pkgs/development/tools/pyre/default.nix setup for ref.

thx @chenrui333 for your research

**Pull request closed without merging**